### DISCUSSION TITLE
Feature Request: Custom File Extension Filter For Code Review

### DISCUSSION THREAD

**Discussion by barnesew** [community/community#8958] [Pull Requests]
## General

Hello. I've been running into a small annoyance while participating in code reviews and I think a simple improvement could be made to help me and others who have a similar situation.

## Problem

We use a lot of code generation in our repository and we like to have it tracked in source control. Each of the files has a file naming convention of "\*.gen.ext". Whenever we make a PR to start a code review, we have no way to filter by the sub-extension; Only direct extensions are present in the file filter dropdown (\*.ts, \*.js, \*.html, etc.) as seen here:

![Screen Shot 2021-12-16 at 11 02 07 AM](https://user-images.githubusercontent.com/18120656/146406193-5034fe6c-8f0c-484a-a2c2-0e4bc5f97782.png)

Since we don't want to view the generated files, it would be nice to have a way to filter them out during reviews.

## Possible Solution

One possible solution could be to allow for a custom file filter in the dropdown where users can specify a wildcard filter kind of like how VSCode handles a global search:

![Screen Shot 2021-12-16 at 11 07 13 AM](https://user-images.githubusercontent.com/18120656/146406828-d199bf80-efd1-4209-8b33-833dd13b1576.png)

Another possible solution could be to include sub-extensions in the filter options such as \*.gen.ts. Since there would be overlap between the sub-extensions and the actual file extensions, this sounds like a less desirable solution.

**Comment by ryanwilliams83** [comment]
@barnesew, I just learned that I can add a `.gitattributes` file to the repo and include `*.gen.* linguist-generated` which helps a little with this problem; but I still desperately need the ability to uncheck them in the file filter drop-down.

https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github
![image](https://github.com/community/community/assets/3871256/30dfdee2-f91b-4c8a-8243-4580de1a3ea0)
